jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -0,0 +1,137 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('user-profile', function() {
+
+  juju.components.UserProfile = React.createClass({
+
+    propTypes: {
+      jem: React.PropTypes.object,
+      listEnvs: React.PropTypes.func,
+      switchEnv: React.PropTypes.func.isRequired,
+      changeState: React.PropTypes.func.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        envList: []
+      };
+    },
+
+    componentWillMount: function() {
+      this._fetchEnvironments();
+    },
+
+    /**
+      Makes a request of JEM or JES to fetch the users availble environments.
+
+      @method _fetchEnvironments
+    */
+    _fetchEnvironments:  function() {
+      var props = this.props;
+      var jem = props.jem;
+      if (jem) {
+        jem.listEnvironments(this._fetchEnvironmentsCallback);
```

Comment:
It might be possible for you to just pass in the listEnvironments method as a prop rather then `jem`.